### DISCUSSION TITLE
Manipulate or change HLS tags/version

### DISCUSSION THREAD

**Discussion by LifelessMuffin** [savonet/liquidsoap#3694] [Ideas]
Hi everyone, 
I was wondering, is there a way to manipulate the HLS playlist's tags?
For example changing the EXT-X-VERSION tag to a different HLS version (now default to 3), perhaps from version 4 being able to use the audio groups and variants such as:

```
#EXT-X-VERSION:4
#EXT-X-INDEPENDENT-SEGMENTS

# AUDIO groups
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="aud1",LANGUAGE="en",NAME="lang",CHANNELS="2",AUTOSELECT=YES,DEFAULT=YES,URI="audio.m3u8"

# Variants
#EXT-X-STREAM-INF:BANDWIDTH=6015000,AVERAGE-BANDWIDTH=6015000,CODECS="avc1.64002a,mp4a.40.2",RESOLUTION=1920x1080,FRAME-RATE=50.000,AUDIO="aud1"
video.m3u8
```

Thanks :D

**Comment by toots** [comment]
The `main` branch now has a `main_playlist_writter`  that you can use to override anything you want in the main playlist: [output.file.hls](https://www.liquidsoap.info/doc-dev/reference.html#output.file.hls)
